flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -943,15 +945,26 @@ protected CleanupJobState jobReachedTerminalState(ExecutionGraphInfo executionGr
         archiveExecutionGraph(executionGraphInfo);
 
         if (terminalJobStatus.isGloballyTerminalState()) {
+            final JobID jobId = executionGraphInfo.getJobId();
             try {
-                jobResultStore.createDirtyResult(
-                        JobResult.createFrom(executionGraphInfo.getArchivedExecutionGraph()));
+                if (jobResultStore.hasCleanJobResultEntry(jobId)) {
+                    log.warn(
```

Comment:
Should this be logged as a warning (at all)? What does this warning mean from user perspective? Does the user need to take any action to address that?

Reply:
I added it because it indicates that something strange is going on in the `JobResultStore` considering that the user could modify files in the directory of the file-based implementation. In general, this warning should never pop up. But if somebody did something wrong with file names, Flink might store the wrong state under that job.